contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2206,12 +2258,21 @@ static List
  * inputs are sanitized to prevent unexpected behaviour.
  */
 void
-exec_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege)
+exec_grantschema_subcmds(const char *schema, const char *rolname, bool is_grant, bool with_grant_option, AclMode privilege, bool is_create_schema)
 {
 	List		*parsetree_list;
 	ListCell	*parsetree_item;
+	const char *prev_current_user;
+	const char *dbo_role;
+	const char *dbname = get_cur_db_name();
+	MigrationMode baseline_mode = is_user_database_singledb(dbname) ? SINGLE_DB : MULTI_DB;
+	dbo_role = get_dbo_role_name_by_mode(dbname, baseline_mode);
+
+	/* Need dbo user to execute the statements. */
+	prev_current_user = GetUserNameFromId(GetUserId(), false);
+	bbf_set_current_user(dbo_role);
```

Comment:
Let's use better API `SetUserIdAndSecContext` to set the role. Also, there might be chances of user getting leaked if an error comes just before resetting the user. Better use TRY/CATCH.